### PR TITLE
[WIP] Rearrange Trezor source files

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

In preparation to update of Trezor library, rearrange Trezor source files.
- Separation of files from trezor lib upstream and wallet-core additions (e.g. CMakeFile) in different folders.
- Trezor files moved from `trezor-crypto/src` to `trezor-crypto/crypto`
No change to the actual content of trezor code files.

## Testing instructions

Unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]
lgtm. besides failing iOS build

**Comment by hewigovens** [comment]
for iOS, two podspecs and one `project.yml` need to be updated (will simplify this soon)

**Comment by catenocrypt** [comment]
iOS build scripts have to be updated; still Draft.

File: swift/Podfile.lock [hunk 1]
```diff
@@ -17,4 +17,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 159625a957d599dfe52b30d250815212818070a7
 
-COCOAPODS: 1.10.1
+COCOAPODS: 1.9.3
```
Comment (hewigovens) [inline swift/Podfile.lock:20]: can revert this change

Comment (catenocrypt) [reply swift/Podfile.lock]: done

**Review by hewigovens — APPROVED** [review]

**Comment by catenocrypt** [comment]
iOS CI OK (references in swift project files updated).
